ruby/example_code/lambda/lambda_basics.rb

```diff
@@ -0,0 +1,295 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# frozen_string_literal: true
+
+# Purpose
+#
+# Shows how to use the AWS SDK for Ruby v3 to manage and invoke AWS Lambda
+# functions.
+
+require "aws-sdk-lambda"
+require "aws-sdk-s3"
+require "aws-sdk-iam"
+require "logger"
+require "json"
+require "zip"
+
+$cloudwatch_client = Aws::CloudWatchLogs::Client.new
+$iam_client = Aws::IAM::Client.new
+
+# snippet-start:[ruby.example_code.ruby.LambdaWrapper.full]
+# snippet-start:[ruby.example_code.ruby.LambdaWrapper.decl]
+class LambdaWrapper
+  attr_accessor :lambda_client
+
+  def initialize
+    @lambda_client = Aws::Lambda::Client.new
+    @logger = Logger.new($stdout)
+    @logger.level = Logger::WARN
+  end
+  # snippet-end:[ruby.example_code.ruby.LambdaWrapper.decl]
+
+  # snippet-start:[ruby.example_code.lambda.setup_iam]
+  # Get an AWS Identity and Access Management (IAM) role.
+  #
+  # @param iam_role_name: The name of the role to retrieve.
+  # @param action: whether to create or destroy the IAM apparatus
```

Comment:
`  # @param action: Whether to create or destroy the IAM apparatus.`

Reply:
Done.